target/ppc/translate/fp-impl.c.inc

```diff
@@ -1473,6 +1297,83 @@ static void gen_stfqx(DisasContext *ctx)
     tcg_temp_free_i64(t1);
 }
 
+/*            Floating-point Load/Store Instructions                         */
+static bool do_lsfpsd(DisasContext *ctx, int rt, int ra, TCGv displ,
+                      bool update, bool store, bool single)
+{
+    TCGv ea;
+    TCGv_i64 t0;
+    REQUIRE_FPU(ctx);
+    if (update && ra == 0) {
+        gen_invalid(ctx);
+        return true;
+    }
+    gen_set_access_type(ctx, ACCESS_FLOAT);
+    t0 = tcg_temp_new_i64();
+    ea = tcg_temp_new();
+    do_ea_calc(ctx, ra, displ, ea);
```

Comment:
This is incorrect for the 'update' versions, which use the contents of gpr[ra] even when ra == 0.

Reply:
`update && ra == 0` is illegal, as handled in lines 1307-1310: https://github.com/PPC64/qemu/blob/04f7b9fdc9ecc4104a7ba063928600c78b3da786/target/ppc/translate/fp-impl.c.inc#L1307-L1310

Reply:
True. I forgot that ra == 0 is invalid for updates, and misinterpreted the ISA where it just says (RA) instead of (RA | 0).